lib/modules/archivist/usercommands/rawApplyDiff.js

```diff
@@ -1,45 +1,19 @@
-exports.acl = ['*'];
-
-exports.execute = function (state, topic, index, diff, cb) {
-	var clientVault = state.archivist.getPrivateVault('client');
-	if (!clientVault) {
-		return state.error(null, 'Client vault not configured in writeOrder.', cb);
-	}
+const ArchivistModuleError = require('../ArchivistModuleError');
+const archivistModule = require('../');
 
-	var topicApi = state.archivist.getTopicApi(topic, clientVault.name);
-	if (!topicApi) {
-		return state.error(
-			null, 'Unable to load topic API on client vault for topic: ' + topic, cb
-		);
-	}
-
-	state.archivist.getValue(topic, index, { optional: false }, function (error, value) {
-		if (error) {
-			return cb(error);
-		}
+exports.acl = ['*'];
 
-		var allowedActors;
+exports.execute = async function (state, topic, index, diff) {
+	const clientVault = archivistModule.getClientVault(state, topic, index);
+	const topicApi =  archivistModule.getTopicApi(state, clientVault, topic, index);
 
-		try {
-			allowedActors = topicApi.shard(value);
-		} catch (apiError) {
-			return state.error(null, apiError, cb);
-		}
+	return archivistModule.getValue(state, topic, index, { optional: false }).then((value) => {
+		const allowedActors = topicApi.shard(value);
 
 		if (!clientVault.operationAllowedForSession(state, topicApi.acl, 'set', allowedActors)) {
-			return state.error(
-				null,
-				'Actor ' + state.actorId + ' not allowed to set diff value.',
-				cb
-			);
-		}
-
-		try {
-			value.applyDiff(diff);
-		} catch (e) {
-			return state.error(null, e, cb);
+			throw new ArchivistModuleError('Actor not allowed to set diff on this topic', state, topic, index);
```

Comment:
Wouldn't you want to know *who* did this?

Reply:
The state contains the actorId and that actorId will be logged.